incubator/hnc/internal/reconcilers/object.go

```diff
@@ -588,14 +588,13 @@ func (r *ObjectReconciler) writeObject(ctx context.Context, log logr.Logger, ins
 		// all descendant namespaces, and this will fail. In order to handle this
 		// case, we try to delete and re-create the rolebinding here
 
-		// We only found this issue with the RoleBinding object, but we *think* this
-		// will also be helpful for other similar objects that end up with the same error
-		// type. If we find out later that this assumption is not true, we can update the
-		// logic here to only deal with RoleBinding.
+		// We don't apply this logic to other objects because if another object has an
+		// ownerReference pointing to the object we're deleting, it could be deleted as
+		// well, which is undesirable.
 
 		// The error type is 'Invalid' after I tested it out with different error types
 		// from https://godoc.org/k8s.io/apimachinery/pkg/api/errors
-		if err != nil && errors.IsInvalid(err) {
+		if err != nil && errors.IsInvalid(err) && inst.GetKind() == "RoleBinding" {
```

Comment:
Technically, we should check the API group as well (`rbac.authorization.k8s.io`) since any group could have something called "RoleBinding" in it. Unlikely yes, but we may as well be precise.

Reply:
done